MERL-709 - Docs: Migration Guide from wp-graphql-gutenberg (#1294)
wpengine/faustjs: internal/faustjs.org/docs/gutenberg/migration-from-wp-graphql-gutenberg.mdx

---
slug: /gutenberg/migration-from-wp-graphql-gutenberg
title: Migration guide from wp-graphql-gutenberg
description: Migration guide from wp-graphql-gutenberg plugin
---

The purpose of this guide is to provide a list of recommended steps to migrate any blocks you have developed from
`wp-graphql-gutenberg` plugin to `wp-graphql-content-blocks` plugin.

## What is wp-graphql-gutenberg?

As explained in this [tutorial](https://developers.wpengine.com/blog/gutenberg-in-headless-wordpress-wpgraphql-gutenberg),

> is an extension for WPGraphQL that adds the blocks to the WPGraphQL schema just like the `wp-graphql-content-blocks`.

Reviewer: ```suggestion
it's an extension for WPGraphQL that adds the blocks to the WPGraphQL schema just like the `wp-graphql-content-blocks`.
```

However there are some key differences that you need to be aware of:

* WPGraphQL Gutenberg gets the blocks registry and sends it in a network request to the WordPress PHP application. So it needs to be synced from time to time.

> * It also allows the blocks to be served as JSON using the `blocksJSON` field. If requesting data this way, the data are not typechecked and you may overfetch data as a side effect.

Reviewer: ```suggestion
* It also allows the blocks to be served as JSON using the `blocksJSON` field. If requesting data this way, the data is not typechecked and you may overfetch data as a side effect.
```

* Block attributes are using their own types, containing different type for deprecated versions. For example:

> ```

Reviewer: ```suggestion
```graphql
```

...on CoreParagraphBlock {
    attributes {
        ...on CoreParagraphBlockAttributes {
            content
    	}
        ...on CoreParagraphBlockDeprecatedV1Attributes {
            content
        }
    }
}
```

* It does not allow blocks to be returned as a flat list so you have to use deeply nested queries to get the list of `innerBlocks` (and this won't nearly solve the issue 100%).

> * `wp-graphql-content-blocks` does not save anything in the database (this is actually a good thing) compared to `wp-graphql-gutenberg`. Therefore it only supports previews when the hitting of preview button.

Reviewer: ```suggestion
* `wp-graphql-content-blocks` does not save anything in the database (this is actually a good thing) compared to `wp-graphql-gutenberg`. Therefore it only supports previews when the "preview" button is hit in the editor.
```

## How do I migrate a block from wp-graphql-gutenberg?

It mainly depends on the way that you queried the blocks using `wp-graphql-gutenberg`. There are two different cases that you have to consider here:

> ### You used the `blocksJSON` to get the blocks data:

Reviewer: ```suggestion
### You used the `blocksJSON` property to get the blocks data:
```

`wp-graphql-content-blocks` does not expose the `blocksJSON` fields because it is problematic to do so.
Getting the data as plain JSON directly from the database completely overrides the principles of GraphQL and ignores the type safety of the system.

> If one of the properties is altered then, you have no guarantees that the GraphQL server with catch them.

Reviewer: ```suggestion
If one of the properties is altered, you have no guarantee that the GraphQL server will catch them.
```

It shows an implementation of the `CoreCode` block using `wp-graphql-gutenberg` and getting the data using [blockJSON].

If you followed one of our the tutorials for [creating a new block](docs/gutenberg/tutorial/create-a-block-from-wordpress-core#53-create-the-block-fragment) from the WordPress Core Blocks you just need to add the following fragment as a new property:

```jsx
CoreCode.fragments = {
  key: `CoreCodeBlockFragment`,
  entry: gql`
    fragment CoreCodeBlockFragment on CoreCode {
      attributes {
        anchor
  		backgroundColor
  		content
  		className
  		gradient
  		style
  		textColor
      }
    }
  `,
};
```

When the `WordPressBlocksViewer`, renders the component, it passes the the whole block data as a property of that block. If your block is designed to accept different properties for attributes and for innerBlocks you may have create a wrapper to forward the properties into the right slots:

```jsx
export default function CoreCode({attributes, children}) {
	const BlockCode = dynamic(() => import('@/components/blocks/core/BlockCode'))

> 	return <BlockCode {...attributes} innerBlocks={children} >

Reviewer: ```suggestion
	return <BlockCode {...attributes} innerBlocks={children}>
```
